pkgs/state-transition/src/transition.zig

```diff
@@ -27,23 +29,41 @@ pub const StateTransitionOpts = struct {
 //     return;
 // }
 
-// not active in PQ devnet0 - zig will automatically prune this from code
-fn process_execution_payload_header(state: *types.BeamState, block: types.BeamBlock) !void {
-    const expected_timestamp = state.genesis_time + block.slot * params.SECONDS_PER_SLOT;
-    if (expected_timestamp != block.body.execution_payload_header.timestamp) {
-        return StateTransitionError.InvalidExecutionPayloadHeaderTimestamp;
+pub fn is_justifiable_slot(finalized: types.Slot, candidate: types.Slot) !bool {
+    if (candidate < finalized) {
+        return StateTransitionError.InvalidJustifiableSlot;
     }
+
+    const delta: f32 = @floatFromInt(candidate - finalized);
+    if (delta <= 5) {
+        return true;
+    }
+    const delta_x2: f32 = @mod(std.math.pow(f32, delta, 0.5), 1);
+    if (delta_x2 == 0) {
+        return true;
+    }
+    const delta_x2_x: f32 = @mod(std.math.pow(f32, delta + 0.25, 0.5), 1);
+    if (delta_x2_x == 0.5) {
+        return true;
+    }
+
+    return false;
 }
 
-pub fn apply_raw_block(allocator: Allocator, state: *types.BeamState, block: *types.BeamBlock, logger: zeam_utils.ModuleLogger) !void {
-    // prepare pre state to process block for that slot, may be rename prepare_pre_state
-    try state.process_slots(allocator, block.slot, logger);
+pub fn apply_raw_block(
+    allocator: Allocator,
+    state: *types.BeamState,
+    block: *types.BeamBlock,
+    opts: StateTransitionOpts,
+) !void {
+    const logger = opts.logger;
+    const transition_timer = zeam_metrics.lean_state_transition_time_seconds.start();
+    defer _ = transition_timer.observe();
 
-    // process block and modify the pre state to post state
+    try state.process_slots(allocator, block.slot, logger);
```

Comment:
why are you explictly calling process slot, process_block does all that, as I said earlier restore back all STF changes except collection of metrics at the right places